### PR TITLE
revise collection api

### PR DESCRIPTION
**Pull request by ro0NL**
continuation of #220

### CODE REVIEW

File: src/Domain/Exception/UnknownCollectionElementException.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\Domain\Exception;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+final class UnknownCollectionElementException extends \RuntimeException implements DomainExceptionInterface
```
Comment (Lctrs) [inline src/Domain/Exception/UnknownCollectionElementException.php]: It should extends [`\OutOfBoundsException`](https://secure.php.net/manual/fr/class.outofboundsexception.php) instead.

```suggestion
final class UnknownCollectionElementException extends \OutOfBoundsException implements DomainExceptionInterface
```

**Comment by Lctrs** [comment]
Just a small suggestion otherwise I'm 👍 .

**Comment by Lctrs** [comment]
I missed it during my first pass but I think you should also update the phpDocs in DomainCollectionInterface.

**Comment by ro0NL** [comment]
if you mean `@return`.. that's implied `mixed` currently. For `@throws` tend to do it all at once sometime.. see #62

**Comment by Lctrs** [comment]
I meant `@throws` indeed.

**Pull request merged**